library/src/animationManager.cxx

```diff
@@ -283,10 +282,87 @@ bool animationManager::LoadAtTime(double timeValue)
   return true;
 }
 
+// ---------------------------------------------------------------------------------
+void animationManager::CycleAnimation()
+{
+  assert(this->Importer);
+  if (this->AvailAnimations <= 0)
+  {
+    return;
+  }
+
+  this->DisableAllAnimation();
+  this->AnimationIndex += 1;
+
+  if (this->AnimationIndex == this->AvailAnimations)
+  {
+    this->AnimationIndex = -1;
+  }
+
+  this->EnableAllAnimation();
```

Comment:
not sure about this ?

Reply:
EnableAllAnimation()/DisableAllAnimation() accommodates index -1 for "All Animations". 

For example, with `InterpolationTest.glb`:
- Once animation is cycled onto "All Animations", the model should enable animations for all 9 cubes.

Reply:
Ok, I understand the logic now.

How about the following:

Remove EnableAllAnimation/DisableAllAnimation and create instead a "EnableOnlyCurrentAnimation" that handles the -1 logic and only enables the one you need and disable the others.

Same result, but naming is much more clear and logical.